client/post-editor/media-modal/detail/detail-item.jsx

```diff
@@ -84,6 +72,63 @@ class EditorMediaModalDetailItem extends Component {
 		);
 	}
 
+	handleOnRestoreClick = () => {
+		const { site, item, onRestore } = this.props;
+		onRestore( site && site.ID, item );
+	};
+
+	renderRestoreButton() {
+		const {
+			item,
+			translate
+		} = this.props;
+
+		//do a simple guid vs url check
+		const guidParts = url.parse( item.guid );
+		const URLParts = url.parse( item.URL );
+
+		if ( guidParts.pathname === URLParts.pathname ) {
```

Comment:
I'm thinking that maybe we could move this to a shared util file because I (and others) might/want to check whether image was updated or not.